code/modules/research/techweb/_techweb.dm

```diff
@@ -217,14 +270,19 @@
 		return FALSE
 	update_node_status(node)
 	if(!force)
-		if(!available_nodes[node.id] || (auto_adjust_cost && (!can_afford(node.get_price(src)))))
+		if(!available_nodes[node.id] || (auto_adjust_cost && (!can_afford(node.get_price(src)))) || !have_experiments_for_node(node))
 			return FALSE
 	if(auto_adjust_cost)
 		remove_point_list(node.get_price(src))
 	researched_nodes[node.id] = TRUE //Add to our researched list
 	for(var/id in node.unlock_ids)
 		visible_nodes[id] = TRUE
-		update_node_status(SSresearch.techweb_node_by_id(id))
+		var/datum/techweb_node/n = SSresearch.techweb_node_by_id(id)
+		if (n.required_experiments.len > 0)
+			add_experiments(n.required_experiments)
+		if (n.discount_experiments.len > 0)
+			add_experiments(n.discount_experiments)
+		update_node_status(n)
```

Comment:
```suggestion
		var/datum/techweb_node/node = SSresearch.techweb_node_by_id(id)
		if (node.required_experiments.len > 0)
			add_experiments(node.required_experiments)
		if (node.discount_experiments.len > 0)
			add_experiments(node.discount_experiments)
		update_node_status(node)
```
This line does not meet our contribution standards, specifically for [variable name compliance](https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names), and should be refactored.